File: backend/app/models/customers.py
"""客户信息与画像模型"""

from sqlalchemy import (
    Column,
    String,
    Boolean,
    Integer,
    ForeignKey,
    Text,
    Index,
    Date,
    Numeric,
)
from sqlalchemy.orm import relationship
from .base import BaseModel


class Customer(BaseModel):
    """客户基础信息表"""

    __tablename__ = "customers"

    company_id = Column(Integer, unique=True, nullable=False, index=True)
    name = Column(String(200), nullable=False, index=True)
    account_type = Column(String(50), index=True)
    price_policy = Column(String(50))
    manager_id = Column(Integer, ForeignKey("users.id"), index=True)
    settlement_cycle = Column(String(20))  # monthly/quarterly/yearly
    settlement_type = Column(String(20), index=True)  # prepaid/postpaid
    is_key_customer = Column(Boolean, default=False, index=True)
    email = Column(String(100), index=True)

    # === 新增字段（方案A） ===
    erp_system = Column(String(100), nullable=True)  # 所属 ERP 系统
    first_payment_date = Column(Date, nullable=True)  # 首次回款时间
    onboarding_date = Column(Date, nullable=True)  # 客户接入时间
    sales_manager_id = Column(
        Integer, ForeignKey("users.id"), nullable=True, index=True
    )  # 销售负责人
    cooperation_status = Column(String(50), nullable=True, index=True, default="active")  # 合作状态：active/suspended/terminated/noused
    is_settlement_enabled = Column(Boolean, nullable=True, default=True)  # 是否启用结算
    is_disabled = Column(Boolean, nullable=True, default=False, index=True)  # 是否停用
    notes = Column(Text, nullable=True)  # 备注

    __table_args__ = (
        Index("idx_customer_sales_manager", "sales_manager_id"),  # 新增
        Index("idx_customer_cooperation_status", "cooperation_status"),  # 新增
        Index("idx_customer_disabled", "is_disabled"),  # 新增
    )

    # 关联
    profile = relationship("CustomerProfile", uselist=False, back_populates="customer")
    balance = relationship("CustomerBalance", uselist=False, back_populates="customer")
    tags = relationship("CustomerTag", back_populates="customer", lazy="selectin")

    def __repr__(self):
        return f"<Customer {self.name}>"


class CustomerProfile(BaseModel):
    """客户画像表"""

    __tablename__ = "customer_profiles"

    customer_id = Column(Integer, ForeignKey("customers.id", ondelete="CASCADE"), unique=True)
    scale_level = Column(String(50))  # 客户规模等级
    consume_level = Column(String(50))  # 客户消费等级
    industry = Column(String(100))
    is_real_estate = Column(Boolean, default=False)
    description = Column(Text)

    # === 新增字段（方案A） ===
    monthly_avg_shots = Column(Integer, nullable=True)  # 月均拍摄量（实际）
    monthly_avg_shots_estimated = Column(Integer, nullable=True)  # 月均拍摄量（测算）
    estimated_annual_spend = Column(Numeric(12, 2), nullable=True)  # 预估年消费
    actual_annual_spend_2025 = Column(Numeric(12, 2), nullable=True)  # 2025年实际消费

    # 关联
    customer = relationship("Customer", back_populates="profile")
    tags = relationship("ProfileTag", back_populates="profile", lazy="selectin")

    def __repr__(self):
        return f"<CustomerProfile {self.customer_id}>"

File: frontend/src/api/customers.ts
import api from './index'

// 获取客户列表
export function getCustomers(params?: {
  page?: number
  page_size?: number
  keyword?: string
  account_type?: string
  industry?: string
  customer_level?: string
  manager_id?: number
  settlement_type?: string
  is_key_customer?: boolean
}) {
  return api.get('/customers', { params })
}

// 获取客户详情
export function getCustomer(id: number) {
  return api.get(`/customers/${id}`)
}

// 创建客户
export function createCustomer(data: {
  company_id: string
  name: string
  account_type?: string
  industry?: string
  customer_level?: string
  price_policy?: string
  manager_id?: number
  settlement_cycle?: string
  settlement_type?: string
  is_key_customer?: boolean
  email?: string
}) {
  return api.post('/customers', data)
}

// 更新客户
export function updateCustomer(
  id: number,
  data: {
    name?: string
    account_type?: string
    industry?: string
    customer_level?: string
    price_policy?: string
    manager_id?: number
    settlement_cycle?: string
    settlement_type?: string
    is_key_customer?: boolean
    email?: string
    erp_system?: string
    first_payment_date?: string
    onboarding_date?: string
    sales_manager_id?: number
    cooperation_status?: string
    is_settlement_enabled?: boolean
    is_disabled?: boolean
    notes?: string
  }
) {
  return api.put(`/customers/${id}`, data)
}

// 删除客户
export function deleteCustomer(id: number) {
  return api.delete(`/customers/${id}`)
}

// 获取客户画像
export function getProfile(customerId: number) {
  return api.get(`/customers/${customerId}/profile`)
}

// 更新客户画像
export function updateProfile(
  customerId: number,
  data: {
    scale_level?: string
    consume_level?: string
    industry?: string
    is_real_estate?: boolean
    description?: string
    monthly_avg_shots?: number
    monthly_avg_shots_estimated?: number
    estimated_annual_spend?: number
    actual_annual_spend_2025?: number
  }
) {
  return api.put(`/customers/${customerId}/profile`, data)
}

// Excel 导入
export function importCustomers(file: File) {
  const formData = new FormData()
  formData.append('file', file)
  return api.post('/customers/import', formData, {
    headers: {
      'Content-Type': 'multipart/form-data',
    },
  })
}

// 下载导入模板
export function downloadImportTemplate() {
  return api.get('/customers/import-template', {
    responseType: 'blob',
  })
}

// Excel 导出
export function exportCustomers(params?: {
  keyword?: string
  account_type?: string
  industry?: string
  customer_level?: string
  manager_id?: number
  settlement_type?: string
  is_key_customer?: boolean
}) {
  return api.get('/customers/export', {
    params,
    responseType: 'blob',
  })
}

// 获取行业类型字典
export function getIndustryTypes() {
  return api.get('/dicts/industry_types')
}
